spec.bs

```diff
@@ -884,14 +931,47 @@ To <dfn>build trusted bidding signals url</dfn> given a [=URL=] |signalsUrl|, an
   [=component percent-encode set=] to |queryParamsList|.
 1. If |keys| is not [=set/is empty|empty=]:
   1. [=list/Append=] "&keys=" to |queryParamsList|.
-  1. Let |keysStr| be the result of [=string/concatenating=] |keys| with separator set to ",".
-  1. [=list/Append=] the result of [=string/UTF-8 percent-encoding=] |keysStr| using
-    [=component percent-encode set=] to |queryParamsList|.
+  1. [=list/Extend=] |queryParamsList| with the result of [=encode trusted signals keys=] with
+    |keys|.
 1. If |igNames| is not [=set/is empty|empty=]:
   1. [=list/Append=] "&interestGroupNames=" to |queryParamsList|.
-  1. Let |igNamesStr| be the result of [=string/concatenating=] |igNames| with separator set to ",".
-  1. [=list/Append=] the result of [=string/UTF-8 percent-encoding=] |igNamesStr| using
-    [=component percent-encode set=] to |queryParamsList|.
+  1. [=list/Extend=] |queryParamsList| with the result of [=encode trusted signals keys=] with
+    |igNames|.
+1. If |experimentGroupId| is not null:
+  1. [=list/Append=] "&experimentGroupId=" to |queryParamsList|.
+  1. [=list/Append=] [=serialize an integer|serialized=] |experimentGroupId| to |queryParamsList|.
+1. Let |fullSignalsUrl| be |signalsUrl|.
+1. Set |fullSignalsUrl|'s [=url/query=] to the result of [=string/concatenating=] |queryParamsList|.
+1. return |fullSignalsUrl|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>build trusted scoring signals url</dfn> given a [=URL=] |signalsUrl|, an [=ordered set=] of
+[=URLs=] |renderUrls|, an [=ordered set=] of [=URLs=] |adComponentRenderUrls|, and an
+{{unsigned short}} |experimentGroupId|:
+1. Let |queryParamsList| be a new [=list/is empty|empty=] [=list=].
+1. [=list/Append=] "hostname=" to |queryParamsList|.
+1. [=list/Append=] the result of [=string/UTF-8 percent-encoding=] [=this=]'s
+  [=relevant settings object=]'s [=environment/top-level origin=] using

```

Comment:
This algorithm is also called from in parallel so we do not have access to `this` or what's currently after it.

Reply:
Done.